### PR TITLE
Sync lints

### PR DESCRIPTION
**Pull request by a14n**
## Description

Sync lint list and enable:

* [exhaustive_cases](https://dart-lang.github.io/linter/lints/exhaustive_cases.html)
* [leading_newlines_in_multiline_strings](https://dart-lang.github.io/linter/lints/leading_newlines_in_multiline_strings.html)

I also :

* replace `default:` with `case XXX` when a single case was left
* remove `default:` when there was no chance to enter it.

## Related Issues

None

## Tests

None

### CODE REVIEW

File: packages/flutter_tools/lib/src/commands/create.dart [hunk 1]
```diff
@@ -1021,6 +1021,7 @@ String _validateProjectDir(String dirPath, { String flutterRoot, bool overwrite
       case FileSystemEntityType.link:
         // Do not overwrite links.
         return "Invalid project name: '$dirPath' - refers to a link.";
+      default:
```
Comment (goderbauer) [inline packages/flutter_tools/lib/src/commands/create.dart:1024]: Why this? If this is intentional, it probably deserves a comment.

File: packages/flutter_tools/lib/src/commands/ide_config.dart [hunk 2]
```diff
@@ -273,6 +273,7 @@ String _validateFlutterDir(String dirPath, { String flutterRoot }) {
       case FileSystemEntityType.link:
         // Do not overwrite links.
         return "Invalid project root dir: '$dirPath' - refers to a link.";
+      default:
```
Comment (goderbauer) [inline packages/flutter_tools/lib/src/commands/ide_config.dart:276]: same.

**Review by goderbauer — APPROVED** [review]
LGTM

**Review by jonahwilliams — APPROVED** [review]
tool changes LGTM

**Pull request merged**